services/nexus/nexus.service.js

```diff
@@ -1,130 +1,221 @@
-            ? `/service/local/lucene/search?g=${groupId}&a=${artifactId}${queryOpt}`
-            : `/service/local/artifact/maven/resolve?r=${repo}&g=${groupId}&a=${artifactId}&v=LATEST${queryOpt}`
-        }`
-
-        request(
-          apiUrl,
-          { headers: { Accept: 'application/json' } },
-          (err, res, buffer) => {
-            if (err != null) {
-              badgeData.text[1] = 'inaccessible'
-              sendBadge(format, badgeData)
-              return
-            } else if (res && res.statusCode === 404) {
-              badgeData.text[1] = 'no-artifact'
-              sendBadge(format, badgeData)
-              return
-            }
-            try {
-              const parsed = JSON.parse(buffer)
-              let version = '0'
-              switch (repo) {
-                case 'r':
-                  if (parsed.data.length === 0) {
-                    badgeData.text[1] = 'no-artifact'
-                    sendBadge(format, badgeData)
-                    return
-                  }
-                  version = parsed.data[0].latestRelease
-                  break
-                case 's':
-                  if (parsed.data.length === 0) {
-                    badgeData.text[1] = 'no-artifact'
-                    sendBadge(format, badgeData)
-                    return
-                  }
-                  // only want to match 1.2.3-SNAPSHOT style versions, which may not always be in
-                  // 'latestSnapshot' so check 'version' as well before continuing to next entry
-                  parsed.data.every(artifact => {
-                    if (isNexusSnapshotVersion(artifact.latestSnapshot)) {
-                      version = artifact.latestSnapshot
-                      return
-                    }
-                    if (isNexusSnapshotVersion(artifact.version)) {
-                      version = artifact.version
-                      return
-                    }
-                    return true
-                  })
-                  break
-                default:
-                  version = parsed.data.baseVersion || parsed.data.version
-                  break
-              }
-              if (version !== '0') {
-                badgeData.text[1] = versionText(version)
-                badgeData.colorscheme = versionColor(version)
-              } else {
-                badgeData.text[1] = 'undefined'
-                badgeData.colorscheme = 'orange'
-              }
-              sendBadge(format, badgeData)
-            } catch (e) {
-              badgeData.text[1] = 'invalid'
-              sendBadge(format, badgeData)
-            }
-          }
-        )
+    if (repo === 'r' || repo === 's') {
+      schema = searchApiSchema
+      url += 'service/local/lucene/search'
+    } else {
+      schema = resolveApiSchema
+      url += 'service/local/artifact/maven/resolve'
+      options.qs.r = repo
+      options.qs.v = 'LATEST'
+    }
+
+    if (queryOpt) {
```

Comment:
It's not obvious from reading this what it's supposed to do. For future reference, stuff like this probably should be its own static method (or function) with a unit test.